pkg/models/alerting/rules/ruler.go

```diff
@@ -591,6 +615,9 @@ func (r *ThanosRuler) UpdateAlertingRules(ctx context.Context, ruleNamespace *co
 		}
 	}
 
+	// The move here is not really move, because the move also requires an update.
```

Comment:
// The move here is not really a move, because the move operation also requires an update.
	// What really happens is that the new rules will be added in other resources first,
	// and then the old rules will be deleted from the original resources.